.github/workflows/tests.yml

```diff
@@ -0,0 +1,36 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+
+env:
+  HATCH_VERSION: "1.14.0"
+  OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 1
```

Comment:
Is there a reason to limit the number parallel jobs?

Reply:
good point. I've added it only to avoid running all jobs at once when testing out, no need to have it now